### DISCUSSION TITLE
What is the proper way to use Deployer with docker containers?

### DISCUSSION THREAD

**Discussion by tomaszkane** [deployphp/deployer#3758] [Help needed]
Hi! I use Deployer 7 to deploy projects to staging server with docker compose installed.

* Containers with PHP 8.0 and PHP 8.2 is always run.
* Task `deploy:vendors` should run on given container.
* But it not possibe?
* So now I `set` container name for `host` and use own `vendors` task to run something like:
```
docker exec {{container_name} bash -c "cd {{container_release_path}} && composer {{composer_action}} {{composer_options}}"
```
I saw that `{{bin/composer}} can be set. But it don't get expected result here? https://github.com/deployphp/deployer/blob/master/recipe/deploy/vendors.php#L31
And also saw this: https://stackoverflow.com/questions/51274489/how-to-use-deployer-with-docker-laradock

**Comment by antonmedv** [comment]
It is not proper way to deploy with Docker. Do not run you commands in container. Build an image.

**Comment by jimwins** [comment]
It may not be the proper way, but I wrote up [how I am using Deployer to deploy to a Docker container](https://trainedmonkey.com/2024/03/09/how_i_use_docker_and_deployer_together).
